cpp/dolfin/common/IndexMap.h

```diff
@@ -100,6 +102,9 @@ class IndexMap
   /// Owner rank (on global communicator) of each ghost entry
   Eigen::Array<std::int32_t, Eigen::Dynamic, 1> ghost_owners() const;
 
+  /// Map from shared indices to sharing processes
+  std::map<std::int32_t, std::set<int>> shared_indices() const;
```

Comment:
I suggest renaming to `compute_shared_indices` (and update docstring) to make clear that it does some work behind the scenes.

Also, update the docstring to make clear what is being returned. E.g. is it just for owned indices?

Reply:
Yes, that is a question I was wondering about. It is only for owned indices, but it would be useful to have the sharing of ghosts too (although that requires further communication). Should I add another function to do that? Or an optional argument? Or just let it be done externally?